scripts/build-bundle

```diff
@@ -49,44 +45,46 @@ tar zxvf ${CHARTS_DIR}/rancher-monitoring-${MONITORING_VERSION}.tgz >/dev/null
-RKE2_IMAGE_TAG=${RKE2_VERSION/+/-} # v1.21.3-rc6+rke2r2 -> v1.21.3-rc6-rke2r2
-sed -i "s,\$RKE2_VERSION,${RKE2_IMAGE_TAG}," ${RANCHERD_IMAGES_DIR}/rancherd-bootstrap-images.txt
-save_image ${RANCHERD_IMAGES_DIR}/rancherd-bootstrap-images.txt ${RANCHERD_IMAGES_DIR}
+image_list_file=${RANCHERD_IMAGES_DIR}/rancherd-bootstrap-images-${VERSION}.txt
+cp ${SCRIPTS_DIR}/images/rancherd-bootstrap-images.txt $image_list_file
+RKE2_VERSION_NORMALIZED=${RKE2_VERSION/+/-} # v1.21.3-rc6+rke2r2 -> v1.21.3-rc6-rke2r2
+sed -i "s,\$RKE2_VERSION,${RKE2_VERSION_NORMALIZED}," ${image_list_file}
+save_image "agent" $BUNDLE_DIR ${image_list_file} ${RANCHERD_IMAGES_DIR}
 
-# Images needed during rancherd cluster-init stage.
-cp ${SCRIPTS_DIR}/images/rancher-images.txt ${IMAGES_DIR}
-save_image ${IMAGES_DIR}/rancher-images.txt ${IMAGES_DIR}
+# Rancher images
+image_list_file=${IMAGES_DIR}/rancher-images-${RANCHER_VERSION}.txt
+cp ${SCRIPTS_DIR}/images/rancher-images.txt $image_list_file
+save_image "common" $BUNDLE_DIR $image_list_file ${IMAGES_DIR}
 
 # RKE2 images
 RKE2_IMAGES_URL="https://github.com/rancher/rke2/releases/download/${RKE2_VERSION}"
-if [ -f "${SCRIPTS_DIR}/images/rke2-images.linux-amd64.txt" ] && [ -f "${SCRIPTS_DIR}/images/rke2-images.linux-amd64.tar.zst" ]; then
-  mv "${SCRIPTS_DIR}/images/rke2-images.linux-amd64.txt" "${IMAGES_DIR}/rke2-images.linux-amd64.txt"
-  mv "${SCRIPTS_DIR}/images/rke2-images.linux-amd64.tar.zst" "${IMAGES_DIR}/rke2-images.linux-amd64.tar.zst"
-else
-  get_url "${RKE2_IMAGES_URL}/rke2-images.linux-amd64.txt" "${IMAGES_DIR}/rke2-images.linux-amd64.txt"
-  get_url "${RKE2_IMAGES_URL}/rke2-images.linux-amd64.tar.zst" "${IMAGES_DIR}/rke2-images.linux-amd64.tar.zst"
-fi
+image_list_file="${IMAGES_DIR}/rke2-images.linux-amd64-${RKE2_VERSION_NORMALIZED}.txt"
+image_archive="${IMAGES_DIR}/rke2-images.linux-amd64-${RKE2_VERSION_NORMALIZED}.tar.zst"
+get_url "${RKE2_IMAGES_URL}/rke2-images.linux-amd64.txt" $image_list_file
+get_url "${RKE2_IMAGES_URL}/rke2-images.linux-amd64.tar.zst" $image_archive
+add_image_list_to_metadata "rke2" $BUNDLE_DIR $image_list_file $image_archive
 
 # exclude SR-IOV images
+image_list_file="${IMAGES_DIR}/rke2-images-multus.linux-amd64-${RKE2_VERSION_NORMALIZED}.txt"
 # save_image_list "rke2-images-multus" "${RKE2_IMAGES_URL}/rke2-images-multus.linux-amd64.txt" ${IMAGES_DIR}/rke2-images-multus.linux-amd64.txt
-
 # Workaround for https://github.com/rancher/rke2/issues/2229
-cat > ${IMAGES_DIR}/rke2-images-multus.linux-amd64.txt <<EOF
+cat > $image_list_file <<EOF
 docker.io/rancher/hardened-cni-plugins:v0.9.1-build20211119
 docker.io/rancher/hardened-multus-cni:v3.7.1-build20211119
 EOF
 
-save_image ${IMAGES_DIR}/rke2-images-multus.linux-amd64.txt ${IMAGES_DIR}
+save_image "rke2" $BUNDLE_DIR $image_list_file ${IMAGES_DIR}
 
 # Harvester images: get image list from harvester chart's values file
-image_list_file='harvester-images.txt'
+image_list_file="harvester-images-${VERSION}.txt"
```

Comment:
The `${VERSION}` here is the installer's version. Maybe what you need is Harvester's version?

Reply:
These images and `harvester-repo-images` are packed in the installer build stage, so it's better to track with the installer version.
Because the image lists and other content might change in the installer repo.